### ISSUE TITLE
clang-tidy: /lib64/libm.so.6: version `GLIBC_2.27' not found

### ISSUE THREAD

**Issue by amurzeau** [microsoft/vscode-cpptools#10019]
### Environment

- OS and version: Windows 10 (but in remote mode to Red Hat 7)
- VS Code: 1.70.2
- C/C++ extension: 1.12.4
- OS and version of remote machine (if applicable): Red Hat 7

### Bug Summary and Steps to Reproduce

Bug Summary:

The cpptools extensions show errors about clang-tidy which can't be run on Red Hat 7 as it requires GLIBC_2.27 for the function `log2f`.
This makes clang-tidy not usable.

The exact error is:
```
<extension path>/LLVM/bin/clang-tidy: /lib64/libm.so.6: version `GLIBC_2.27` not found (required by <extension path>/LLVM/bin/clang-tidy)
```

Steps to reproduce:
1. Run the extension on a Red Hat 7 (or Centos 7) machine
2. Open a C/C++ project with this setting enabled: `C_Cpp.codeAnalysis.clangTidy.enabled`
3. Show the output `Log (Remote Serer)`
4. See the above error: `clang-tidy: /lib64/libm.so.6: version GLIBC_2.27 not found`

### Additional Information

Running objdump -T on clang-tidy shows which function is the issue:
```
[...]
0000000000000000      DF *UND*	0000000000000000 (GLIBC_2.27) log2f
[...]
```

**Comment by H-G-Hristov** [comment]
Does clang-tidy support remote scenarios?

**Comment by amurzeau** [comment]
yes it works fine when clang-tidy can run on the remote machine (with a new enough glibc).
I can find the LLVM/bin/clang-tidy process in `ps` on the remote machine. The clang-tidy issues are reported as expected.

As a comparison, here is a working clang-tidy on a Debian Unstable remote, issues are detected and displayed:
![clang-tidy-working](https://user-images.githubusercontent.com/5435069/196056571-a0dd6224-1875-47b7-936a-64a347a7a156.png)

And here, using a Centos 7 remote, there is no issues detected and the error appear in the C/C++ output (requires "Information" log level):

![clang-tidy-centos-failure](https://user-images.githubusercontent.com/5435069/196056606-efab9b28-7eb4-45c8-acae-c5349a20ff74.png)

**Comment by sean-mcmanus** [comment]
The fix is available with 1.13.3 (pre-release) for Linux x64 or arm64 (or Alpine): https://github.com/microsoft/vscode-cpptools/releases/tag/v1.13.3

However, Linux arm 32-bit still builds with shared libraries because otherwise it was crashing. We could open another issue track that if someone is hitting that.
